cdap-common/src/main/java/co/cask/cdap/common/logging/LoggingContextAccessor.java

```diff
@@ -39,15 +41,30 @@
    * </p>
    * @param context context to set
    */
-  public static void setLoggingContext(LoggingContext context) {
+  public static ThreadMDCContext setLoggingContext(LoggingContext context) {
```

Comment:
@chtyim If we were to use Cancellable, wouldn't we have to keep the old context inside the class? In that case if a thread calls set(1), set(2), reset(2), reset(1), context from 1 would be lost. Please let me know your thoughts.